### PR TITLE
Corrections and R-package tests for RCoT

### PR DESCRIPTION
**Pull request by ewengillies**
TODO : change back the _set_max_condition_dim() function and the changes to the test code

### CODE REVIEW

File: tigramite/pcmci.py [hunk 1]
```diff
@@ -761,12 +757,12 @@ def _set_max_condition_dim(self, max_conds_dim, tau_min, tau_max):
 
         Returns
         -------
-        max_cond_dim : int
+        max_conds_dim : int
             Input maximum condition dimension or default
         """
         # Check if an input was given
         if max_conds_dim is None:
-            max_conds_dim = self.N * (tau_max - tau_min + 1)
```
Comment (ewengillies) [inline tigramite/pcmci.py:769]: remove +1 here, make sure that max_conds_dim = 2 iterates through 0, 1, 2 in main loop and wherever similar concept is used.

File: install_r_packages.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Install rpy2 and some dependencies
+conda install rpy2
+# Install r-MASS package, which is a dependency of RCIT/RCOT
+R -e 'install.packages("MASS", repos = "https://cloud.r-project.org/")' 
+# Install momentchi2, whichs is a dependency of RCIT/RCOT
+R -e 'install.packages("momentchi2", repos = "https://cloud.r-project.org/")' 
+# Install devtools so we can install from github
+R -e 'install.packages("devtools", repos = "https://cloud.r-project.org/")' 
+# Install RCOT/RCIT
+R -e 'library(devtools); devtools::install_github("ericstrobl/RCIT")'
```
Comment (ewengillies) [inline install_r_packages.sh:10]: download RCIT package into tigramite distribution. maybe git repo inside a git repo?

File: setup.py [hunk 3]
```diff
@@ -3,6 +3,8 @@
 from setuptools import setup, Extension
 from setuptools.command.build_ext import build_ext
 
+extra_link_args=['-L/usr/lib/x86_64-linux-gnu/']
```
Comment (ewengillies) [inline setup.py:6]: maybe remove this.

File: setup.py [hunk 4]
```diff
@@ -30,21 +32,18 @@ def define_extension(extension_name, source_files=None):
...
+    # If we are, try to import and use cythonize
+    try:
+        from Cython.Build import cythonize
+        # Replace any extension in the source file list with .pyx
+        source_files = [".".join(f.split(".")[:-1] + ["pyx"]) \
+                        for f in source_files]
+        # Return the cythonized extension
+        return cythonize(Extension(extension_name, source_files, extra_link_args=extra_link_args))
+    except ImportError:
+        print("Cython cannot be found.  Skipping compilation of cython code"+\
```
Comment (ewengillies) [inline setup.py:44]: edit this error message, avoid the usage of compilation.

File: setup.py (hunk 3 above)
Comment (ewengillies) [reply setup.py:6]: done

File: setup.py (hunk 4 above)
Comment (ewengillies) [reply setup.py:44]: done

File: tigramite/pcmci.py (hunk 1 above)
Comment (ewengillies) [reply tigramite/pcmci.py:769]: done

**Review by jakobrunge — APPROVED** [review]

**Pull request merged**